cmake/setup.py.cmake

```diff
@@ -27,26 +31,27 @@ if(BUILD_PYTHON)
 
   # Create setup.py
   set(SETUP_PY_IN "${CMAKE_CURRENT_LIST_DIR}/setup.py.in")
-  set(SETUP_PY "${PYTHON_DEST}/setup.py")
+  set(SETUP_PY "${CMAKE_CURRENT_BINARY_DIR}/cmake/setup.py")
   set(SETUP_PY_INIT "${PYTHON_DEST}/sirf/__init__.py")
-  message(STATUS "setup.py:${SETUP_PY}")
   configure_file("${SETUP_PY_IN}" "${SETUP_PY}")
+  install(FILES "${SETUP_PY}" DESTINATION "${PYTHON_DEST}")
+  message(STATUS "setup.py:${PYTHON_DEST}/setup.py")
 
   if(PYTHONINTERP_FOUND)
     # python setup.py build
     add_custom_command(OUTPUT "${SETUP_PY_INIT}"
       COMMAND "${CMAKE_COMMAND}" -E make_directory "${PYTHON_DEST}/sirf"
       COMMAND "${CMAKE_COMMAND}" -E touch "${SETUP_PY_INIT}"
-      COMMAND "${PYTHON_EXECUTABLE}" setup.py build
-      DEPENDS "${SETUP_PY}"
+      # COMMAND "${PYTHON_EXECUTABLE}" setup.py build
```

Comment:
This command now only makes `PYTHON_DEST/sirf` (but that's now already done above at install-time), and creates a zero-length `PYTHON_DEST/__init__py`. However, it's then added as a dependency on `ALL` below. That to me means that it's run at build-time anyway. But that was what we try to avoid?

Seems to me we need to either use `configure_file` to create an empty `__init__.py` (seems to me the preferred option), or add
```cmake
 install(CODE "execute_process(COMMAND\n\
    \"${CMAKE_COMMAND}\" -E touch \"${SETUP_PY_INIT}\")")
```

Reply:
fixed